src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -128,18 +128,6 @@ function VisEditor(
   // SearchSource is a promise-based stream of search results that can inherit from other search sources.
   const { vis, searchSource } = savedVis;
 
-  // adds top level search source to the stack to which global filters are applied
-  const getTopLevelSearchSource = (searchSource) => {
-    if (searchSource.getParent()) return getTopLevelSearchSource(searchSource.getParent());
-    return searchSource;
-  };
-
-  const topLevelSearchSource =  getTopLevelSearchSource(searchSource);
-  const globalFiltersSearchSource = searchSource.create();
-  globalFiltersSearchSource.setField('index', searchSource.getField('index'));
```

Comment:
mess with the search sources is removed, we should pass global filters in the same way we do in dashboards

Reply:
btw, whole `changes in kibana` commit is extracted into separate PRs and will be removed once those are merged and this PR will be rebased.